app/src/main/java/com/bnyro/translate/api/deepl/DeeplEngine.kt

```diff
@@ -33,43 +48,124 @@ class DeeplEngine : TranslationEngine(
 ) {
     val useFreeApiKey = this.name + "selectedApiVersion"
     private val nonFreeUrl = "https://api.deepl.com"
+    private val webUrl = "https://www2.deepl.com"
     private lateinit var api: DeepL
     override fun createOrRecreate(): TranslationEngine = apply {
         api = RetrofitHelper.createApi(this)
     }
 
     private val apiKeyString = "DeepL-Auth-Key " + getApiKey()
 
-    override suspend fun getLanguages(): List<Language> = api.getLanguages(
-        apiKeyString
-    ).map {
-        Language(
-            code = it.language.lowercase(),
-            name = it.name
-        )
-    }.toMutableList().apply {
-        add(
-            0,
+    override suspend fun getLanguages(): List<Language> = if (getApiKey().isNotEmpty()) {
+        api.getLanguages(
+            apiKeyString
+        ).map {
             Language(
-                "",
-                "Auto"
+                code = it.language.lowercase(),
+                name = it.name
             )
+        }.toMutableList().apply {
+            add(
+                0,
+                Language(
+                    "",
+                    "Auto"
+                )
+            )
+        }
+    } else {
+        listOf(
+            Language("", "Auto"),
+            Language("ar", "Arabic"),
+            Language("bg", "Bulgarian"),
+            Language("zh", "Chinese"),
+            Language("cs", "Czech"),
+            Language("da", "Danish"),
+            Language("nl", "Dutch"),
+            Language("en", "English"),
+            Language("et", "Estonian"),
+            Language("fi", "Finnish"),
+            Language("fr", "French"),
+            Language("de", "German"),
+            Language("el", "Greek"),
+            Language("hu", "Hungarian"),
+            Language("id", "Indonesian"),
+            Language("it", "Italian"),
+            Language("ja", "Japanese"),
+            Language("ko", "Korean"),
+            Language("lv", "Latvian"),
+            Language("lt", "Lithuanian"),
+            Language("nb", "Norwegian (bokmål)"),
+            Language("pl", "Polish"),
+            Language("pt", "Portuguese"),
+            Language("ro", "Romanian"),
+            Language("ru", "Russian"),
+            Language("sk", "Slovak"),
+            Language("sl", "Slovenian"),
+            Language("es", "Spanish"),
+            Language("sv", "Swedish"),
+            Language("tr", "Turkish"),
+            Language("uk", "Ukrainian")
```

Comment:
Where did you extract these languages from? The website? It'd be nice to add a comment here so that future people know where to look when updating the supported languages.